src/Controllers/ResultController.php

```diff
@@ -18,27 +27,45 @@ public function setService(AbstractService $resultService)
         $this->resultService = $resultService;
     }
 
+    /**
+     * This function renders for admin all the quizzes that've been taken by users.
+     * It calls the resultService for QuizInstanceDTOs which contain
+     * userName, quizName, and the Score of the user taken quiz.
+     * if the process is successful, the Renderer gets those DTOs, else
+     * the rendered will just display and empty results page
+     * @return \Framework\Http\Response
+     */
     public function getAllUserTest(RequestInterface $request)
     {
+        $role = $this->session->get('role');
+
+        //Only admins ca see all ever taken quizzes
+        if ($role !== 'admin') {
+            return $this->getRedirectPage("http://local.quiz.com/error/404");
+        }
```

Comment:
duplicate, you rewrite it in many methods, move it to a dedicated method

Reply:
also, I said a number of times that, in my opinion, services should not know about the Request object. what if you want to use the same service in a CLI tool?